docs/tutorials/gluon/performance.md

```diff
@@ -0,0 +1,483 @@
+benchmark_network(data, label, net, loss_fn, trainer)
+```
+
+    
+     ### benchmark_dataloader 
+    
+    ........................
+    
+    total startup time: 0.1723
+    average iterations/sec: 6.1231
+    average samples/sec: 97.9701
+    
+     ### benchmark_network 
+    
+    ........................
+    
+    average iterations/sec: 13.6279
+    average samples/sec: 218.0460
+
+
+Our data loading pipeline appears to be the bottleneck for training: ~100 samples/second compared with ~200 samples/second for network execution. One limiting factor could be disk throughput when reading samples (using a SSD instead of HDD can help with this), but in this case we intentionally added a delay in data transformation. Augmentation can often be a bottleneck in training if the following trick isn't applied.
+
+## Tips & Tricks #1: Use multiple workers on `DataLoader`
+
+In the previous section, we established that the data loading component of the training loop was the bottleneck. Instead of simply removing the artificial delay, let's assume it was some pre-processing or augmentation step that couldn't be removed. We found that the CPU utilization was fixed at 100%, but this was just for a single core. Usually machines have multiple cores and with one easy trick we can leverage more CPU cores to pre-process the data. Setting `num_workers` on the `DataLoader` will result in multiple workers being used to preprocess the data. We can use `multiprocessing.cpu_count()` to find the number of CPU cores available on the machine, and we save 1 core for the main thread.
+
+
+```python
+num_workers = multiprocessing.cpu_count() - 1
+dataloader = mx.gluon.data.DataLoader(dataset,
+                                      batch_size=batch_size,
+                                      shuffle=True,
+                                      last_batch="discard",
+                                      num_workers=num_workers)
+print('Using {} workers for DataLoader.'.format(num_workers))
+```
+
+    Using 7 workers for DataLoader.
+
+
+We benchmark the two main components once again:
+
+
+```python
+print('\n', '### benchmark_dataloader', '\n')
+benchmark_dataloader(dataloader)
+print('\n', '### benchmark_network', '\n')
+data, label = next(iter(dataloader))
+data = data.as_in_context(ctx)
+label = label.as_in_context(ctx)
+benchmark_network(data, label, net, loss_fn, trainer, iters=10)
+```
+
+    
+     ### benchmark_dataloader 
+    
+    ........................
+    
+    total startup time: 0.1967
+    average iterations/sec: 45.6467
+    average samples/sec: 730.3466
+    
+     ### benchmark_network 
+    
+    .........
+    
+    average iterations/sec: 13.2545
+    average samples/sec: 212.0723
+
+
+Our data loading pipeline is no longer the bottleneck for training throughput: ~700 samples per second versus ~200 samples per second for network execution as before. We can now focus our attention on improving the network throughput.
```

Comment:
Maybe you could add a mention here about [NVIDIA DALI](https://github.com/NVIDIA/DALI), which has MXNet integration and could be useful especially for multi GPU systems?

Reply:
Added.